diff --git a/src/dawn/native/CMakeLists.txt b/src/dawn/native/CMakeLists.txt
index efa040a3ad..01a7641a54 100644
--- a/src/dawn/native/CMakeLists.txt
+++ b/src/dawn/native/CMakeLists.txt
@@ -306,7 +306,6 @@ list(APPEND conditional_private_platform_depends
 if (WIN32 AND NOT WINDOWS_STORE)
   list(APPEND conditional_private_platform_depends
        user32.lib
-       onecore_apiset.lib
   )
 endif()
 
