diff --git a/src/dawn/native/CMakeLists.txt b/src/dawn/native/CMakeLists.txt
index efa040a3ad..626b194eee 100644
--- a/src/dawn/native/CMakeLists.txt
+++ b/src/dawn/native/CMakeLists.txt
@@ -1024,6 +1024,7 @@ if (DAWN_BUILD_MONOLITHIC_LIBRARY)
         dawn_public_config
       PRIVATE
         ${conditional_private_platform_depends}
+        dxcompiler
     )
 
     set(webgpu_dawn_public_headers)
diff --git a/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp b/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp
index 21489b4c5f..dbadffd862 100644
--- a/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp
+++ b/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp
@@ -25,6 +25,7 @@
 // OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#define DxcCreateInstance DxcCreateInstance_FromWindowsHeadersWhatever
 #include "src/dawn/native/d3d12/PlatformFunctionsD3D12.h"
 
 #include <array>
@@ -34,11 +35,20 @@
 
 #include "src/dawn/common/SystemUtils.h"
 #include "src/utils/log.h"
+#undef DxcCreateInstance
+
+extern "C" HRESULT WINAPI DXC_Init(void);
+extern "C" HRESULT WINAPI DXC_Done(void);
+extern "C" HRESULT WINAPI DxcCreateInstance(REFCLSID rclsid, REFIID riid, LPVOID* ppv);
 
 namespace dawn::native::d3d12 {
 
 PlatformFunctions::PlatformFunctions() = default;
-PlatformFunctions::~PlatformFunctions() = default;
+PlatformFunctions::~PlatformFunctions() {
+    if (mDxcLib) {
+        DXC_Done();
+    }
+}
 
 MaybeError PlatformFunctions::Initialize(std::span<const std::string> searchPaths) {
     DAWN_TRY(Base::Initialize());
@@ -185,30 +195,18 @@ MaybeError PlatformFunctions::EnsureDXCLibraries(std::span<const std::string> se
     // LoadDXIL and LoadDXCompiler will fail in UWP, but Initialize() can still be
     // successfully executed.
 
-    if (mDXILLib.Valid()) {
+    if (mDxcLib) {
         // The libraries are already loaded, no need to load them again.
-        DAWN_CHECK(mDXCompilerLib.Valid());
         return {};
     }
 
-    DynamicLib dxilLib;
-    std::string error;
-    // DXIL must be loaded before DXC, otherwise shader signing is unavailable
-    if (!dxilLib.Open("dxil.dll", searchPaths, &error)) {
-        return DAWN_INTERNAL_ERROR(std::move(error));
+    if (FAILED(DXC_Init())) {
+        return DAWN_INTERNAL_ERROR("DXC_Init failed");
     }
 
-    DynamicLib dxCompilerLib;
-    if (!dxCompilerLib.Open("dxcompiler.dll", searchPaths, &error)) {
-        return DAWN_INTERNAL_ERROR(std::move(error));
-    }
+    mDxcLib = true;
+    dxcCreateInstance = &DxcCreateInstance;
 
-    if (!dxCompilerLib.GetProc(&dxcCreateInstance, "DxcCreateInstance", &error)) {
-        return DAWN_INTERNAL_ERROR(std::move(error));
-    }
-
-    mDXCompilerLib = std::move(dxCompilerLib);
-    mDXILLib = std::move(dxilLib);
     return {};
 }
 #endif  // DAWN_USE_BUILT_DXC
diff --git a/src/dawn/native/d3d12/PlatformFunctionsD3D12.h b/src/dawn/native/d3d12/PlatformFunctionsD3D12.h
index cdb82188a6c..bc426d0efdb 100644
--- a/src/dawn/native/d3d12/PlatformFunctionsD3D12.h
+++ b/src/dawn/native/d3d12/PlatformFunctionsD3D12.h
@@ -114,4 +114,3 @@ class PlatformFunctions final : public d3d::PlatformFunctions {
     DynamicLib mPIXEventRuntimeLib;
-    DynamicLib mDXILLib;
-    DynamicLib mDXCompilerLib;
+    bool mDxcLib = false;
 
