diff --git a/src/dawn/native/CMakeLists.txt b/src/dawn/native/CMakeLists.txt
index efa040a3ad..626b194eee 100644
--- a/src/dawn/native/CMakeLists.txt
+++ b/src/dawn/native/CMakeLists.txt
@@ -1024,6 +1024,7 @@ if (DAWN_BUILD_MONOLITHIC_LIBRARY)
         dawn_public_config
       PRIVATE
         ${conditional_private_platform_depends}
+        dxcompiler
     )
 
     set(webgpu_dawn_public_headers)
diff --git a/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp b/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp
index 81d85f9767..eca75e59f3 100644
--- a/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp
+++ a/src/dawn/native/d3d12/PlatformFunctionsD3D12.cpp
@@ -25,6 +25,7 @@
 // OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#define DxcCreateInstance DxcCreateInstance_FromWindowsHeadersWhatever
 #include "src/dawn/native/d3d12/PlatformFunctionsD3D12.h"
 
 #include <array>
@@ -34,11 +35,20 @@
 
 #include "src/dawn/common/SystemUtils.h"
 #include "src/utils/log.h"
+#undef DxcCreateInstance
+
+extern "C" HRESULT WINAPI DXC_Init(void);
+extern "C" HRESULT WINAPI DXC_Done(void);
+extern "C" HRESULT WINAPI DxcCreateInstance(REFCLSID rclsid, REFIID riid, LPVOID* ppv);
 
 namespace dawn::native::d3d12 {
 
 PlatformFunctions::PlatformFunctions() = default;
-PlatformFunctions::~PlatformFunctions() = default;
+PlatformFunctions::~PlatformFunctions() {
+    if (mDxcLib) {
+        DXC_Done();
+    }
+}
 
 MaybeError PlatformFunctions::Initialize(std::span<const std::string> searchPaths) {
     DAWN_TRY(Base::Initialize());
@@ -202,6 +212,17 @@ void PlatformFunctions::LoadPIXRuntime(std::span<const std::string> searchPaths)
 
 #if defined(DAWN_USE_BUILT_DXC)
 MaybeError PlatformFunctions::EnsureDXCLibraries(std::span<const std::string> searchPaths) {
+#if 1
+    if (mDxcLib) {
+         return {};
+     }
+    if (FAILED(DXC_Init())) {
+        return DAWN_INTERNAL_ERROR("DXC_Init failed");
+    }
+    mDxcLib = true;
+    dxcCreateInstance = &DxcCreateInstance;
+    return {};
+#else
     // TODO(dawn:766)
     // Statically linked with dxcompiler.lib in UWP
     // currently linked with dxcompiler.lib making CoreApp unable to activate
@@ -225,6 +246,7 @@ MaybeError PlatformFunctions::EnsureDXCLibraries(std::span<const std::string> se
 
     mDXCompilerLib = std::move(dxCompilerLib);
     return {};
+#endif
 }
 #endif  // DAWN_USE_BUILT_DXC
 }  // namespace dawn::native::d3d12
diff --git a/src/dawn/native/d3d12/PlatformFunctionsD3D12.h b/src/dawn/native/d3d12/PlatformFunctionsD3D12.h
index cdb82188a6c..bc426d0efdb 100644
--- a/src/dawn/native/d3d12/PlatformFunctionsD3D12.h
+++ b/src/dawn/native/d3d12/PlatformFunctionsD3D12.h
@@ -112,7 +112,7 @@ class PlatformFunctions final : public d3d::PlatformFunctions {
     DynamicLib mD3D12Lib;
     DynamicLib mD3D11Lib;
     DynamicLib mPIXEventRuntimeLib;
-    DynamicLib mDXCompilerLib;
+    bool mDxcLib = false;
 
 #ifdef DAWN_USE_AGILITY_SDK
     // Called once by Initialize().
